vcd/resource_vcd_vapp_vm.go

```diff
@@ -904,7 +904,10 @@ func resourceVmHotUpdate(d *schema.ResourceData, meta interface{}, vmType typeOf
 	}
 
 	// due the bug in VCD 10.1 hot update possible for adding new or update existing network, removing of network has to be done with cold update
-	if d.HasChange("network") && !isNetworkRemovedInVcd101(d, meta) {
+	// There are a few cases when hot NIC updates are not possible:
+	// * VCD 10.1 does not allow to remove NICs
+	// * Primary NIC cannot be removed on a powered on VM
+	if d.HasChange("network") && !isNetworkRemovedInVcd101(d, meta) && !isPrimaryNicRemoved(d) {
 		networkConnectionSection, err := networksToConfig(d, vdc, vapp, vcdClient)
 		if err != nil {
 			return fmt.Errorf("unable to setup network configuration for update: %s", err)
```

Comment:
Does the user see a descriptive enough error to understand that removal failed, because VM is powered on?